lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -269,6 +269,19 @@ void FrontendInputsAndOutputs::addPrimaryInputFile(StringRef file,
   addInput(InputFile(file, true, buffer));
 }
 
+void FrontendInputsAndOutputs::overrideBufferForInput(
+    StringRef file, llvm::MemoryBuffer *buffer) {
+  // FIXME: If we knew this was only interesting for primary inputs, we could
+  // use the PrimaryInputsByName map instead of a linear scan. But at the time
+  // this was added, the CompilerInvocation used for code completion does not
+  // guarantee that.
+  auto indexIter = llvm::find_if(AllInputs, [file](const InputFile &input) {
```

Comment:
When can code-completion be on a non-primary input?  That's surprising to me.

Reply:
Oh, I didn't realize/remember we were building driver-to-frontend command lines that recorded the code completion file as a primary input. If we can rely on that then great.

Reply:
Hmm, looking at `resolveSymbolicLinksInInputs` we can't quite *assert* that this is the case, but I'd be okay with failing the request if it somehow isn't primary since it seems like something must have gone really wrong.

Reply:
Okay, that'll definitely simplify things!

Reply:
Oh shoot, this may not be true for `swift-ide-test`.

Reply:
Okay, I think if swift-ide-test's `doCodeCompletion` changes to call `addPrimaryInput` it should work.

Reply:
Yeah, we can do whatever we want to `swift-ide-test`. It's the actual completion through SourceKit I was worried about.

(I'm also confused why SourceKit is resolving symbolic links separately from whatever the compiler does. Not only does that seem inconsistent, but you might actually get different output because the filename is different.)